deepmd/train/trainer.py

```diff
@@ -597,69 +654,43 @@ def _build_network(self, data, suffix=""):
 
     def _build_training(self):
         trainable_variables = tf.trainable_variables()
-        if self.run_opt.is_distrib:
-            if self.scale_lr_coef > 1.0:
-                log.info("Scale learning rate by coef: %f", self.scale_lr_coef)
-                optimizer = tf.train.AdamOptimizer(
-                    self.learning_rate * self.scale_lr_coef
-                )
-            else:
-                optimizer = tf.train.AdamOptimizer(self.learning_rate)
-            optimizer = self.run_opt._HVD.DistributedOptimizer(optimizer)
-        else:
-            optimizer = tf.train.AdamOptimizer(learning_rate=self.learning_rate)
-        if self.mixed_prec is not None:
-            _TF_VERSION = Version(TF_VERSION)
-            # check the TF_VERSION, when TF < 1.12, mixed precision is not allowed
-            if _TF_VERSION < Version("1.14.0"):
-                raise RuntimeError(
-                    "TensorFlow version %s is not compatible with the mixed precision setting. Please consider upgrading your TF version!"
-                    % TF_VERSION
-                )
-            elif _TF_VERSION < Version("2.4.0"):
-                optimizer = tf.train.experimental.enable_mixed_precision_graph_rewrite(
-                    optimizer
-                )
-            else:
-                optimizer = tf.mixed_precision.enable_mixed_precision_graph_rewrite(
-                    optimizer
-                )
         if not self.multi_task_mode:
+            if self.run_opt.is_distrib:
+                if self.scale_lr_coef > 1.0:
+                    log.info("Scale learning rate by coef: %f", self.scale_lr_coef)
+                    optimizer = tf.train.AdamOptimizer(
+                        self.learning_rate * self.scale_lr_coef
+                    )
+                else:
+                    optimizer = tf.train.AdamOptimizer(self.learning_rate)
+                optimizer = self.run_opt._HVD.DistributedOptimizer(optimizer)
+            else:
+                optimizer = tf.train.AdamOptimizer(learning_rate=self.learning_rate)
+            if self.mixed_prec is not None:
+                _TF_VERSION = Version(TF_VERSION)
+                # check the TF_VERSION, when TF < 1.12, mixed precision is not allowed
+                if _TF_VERSION < Version("1.14.0"):
+                    raise RuntimeError(
+                        "TensorFlow version %s is not compatible with the mixed precision setting. Please consider upgrading your TF version!"
+                        % TF_VERSION
+                    )
+                elif _TF_VERSION < Version("2.4.0"):
+                    optimizer = (
+                        tf.train.experimental.enable_mixed_precision_graph_rewrite(
+                            optimizer
+                        )
+                    )
+                else:
+                    optimizer = tf.mixed_precision.enable_mixed_precision_graph_rewrite(
+                        optimizer
+                    )
+
```

Comment:
This is actually what `_build_optimizer` is. Can you extract it to a new method?